tool_collections/bamtools/bamtools_filter/bamtools-filter.xml

```diff
@@ -1,322 +1,299 @@
-
------
-
-**Example 3. Complex filtering with multiple conditions**
-
-Suppose now you would like to select **either** reads that (**1**) have (*1.1*) no mismatches and (*1.2*) are on the forward strand **OR** (**2**) reads that have (*2.1*)
-at least one mismatch and (*2.2*) are on the reverse strand. In this scenario we have to set up two conditions: (**1**) and (**2**) each with two filters: *1.1* and *1.2* as well as *2.1* and *2.2*.
-The following screenshot expalins how this can be done:
-
-.. image:: complex-filters.png
-
------
-
-**Example 4. Even more complex filtering with Rules**
-
-In the above example we have used two conditions (Condition 1 and Condition 2). Using multiple conditions allows to combine them and a variety of ways to enable even more powerful filtering.
-For example, suppose get all reads that (**1**) do NOT map to mitochondria and either (**2**) have mapping quality over 20, or (**3**) are in properly mapped pairs. The logical rule to enable such
-filtering will look like this::
-
- !(1) &amp; (2 | 3)
-
-Here, numbers 1, 2, and 3 represent conditions. The following screenshot illustrates how to do this in Galaxy:
-
-.. image:: rule.png
-
-There are three conditions here, each with a single filter. A text entry area that can be opened by clicking on the **Would you like to set rules?** checkbox enables you to enter a rule.
-Here numbers correspond to numbers of conditions as they are shown in the interface. E.g., 1 corresponds to condition 1, 2 to condition 2 and so on... In human language this means::
-
- NOT condition 1 AND (condition 2 OR condition 3)
-
------
-
-**JSON script file**
-
-This tool produces two outputs. One of the them is a BAM file containing filtered reads. The other is a JSONified script. It can help you to see how your instructions are sent to BAMTools.
-For instance, the example 4 looks like this in the JSON form::
-
-       {
-        "filters":
-        [
-          { "id": "1",
-            "tag":"NM:=0",
-            "isReverseStrand":"false"
-          },
-          { "id": "2",
-            "tag":"NM:>0",
-            "isReverseStrand":"true"
-          }
-        ]
-      }
-
-
------
-
-**More information**
-
-.. class:: infomark
-
-Additional information about BAMtools can be found at https://github.com/pezmaster31/bamtools/wiki
-
-
-    </help>
-    <citations>
-        <citation type="doi">10.1093/bioinformatics/btr174</citation>
-    </citations>
-</tool>
+<tool id="bamFilter" name="Filter" version="2.4.1">
+    <description>BAM datasets on a variety of attributes</description>
+    <requirements>
+        <requirement type="package" version="2.4.0">bamtools</requirement>
+    </requirements>
+    <command>
+        <![CDATA[
+            cat $script_file > $out_file2;
+            ln -s "${input_bams}" "localbam.bam" &&
+            ln -s "${input_bams.metadata.bam_index}" "localbam.bam.bai" &&
+            bamtools filter -script $script_file -in "localbam.bam" -out "$out_file1"
```

Comment:
Use single quotes around `${input_bams}`, `${input_bams.metadata.bam_index}` and `$out_file1`.
Add single quotes around `$script_file` (twice) and `$out_file2`.
You can also remove quotes around `localbam.bam` and `localbam.bam.bai`.